src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -571,9 +581,13 @@ public void terminateTask(String taskId, TerminateInfo terminateInfo) {
                     default -> TERMINATE_EXCEPTION;
                 };
                 setSystemUserTaskActionAttributes(task, taskAction);
+
                 //Perform Camunda updates
                 camundaService.deleteCftTaskState(taskId);
+                terminationProcessHelper.fetchTerminationProcessFromCamunda(taskId)
```

Comment:
As discussed, before we parse the termination reason from Camunda we need to check if  is already populated, the task object should already have this information.